### PR TITLE
perf(snuba): add orjson experiment

### PR DESCRIPTION
**Pull request by anonrig**
Adds an experiment for replacing JSON parser with `orjson`

Ref: https://github.com/getsentry/sentry/issues/68903

### CODE REVIEW

File: src/sentry/snuba/tasks.py [hunk 1]
```diff
@@ -250,14 +252,23 @@ def _create_snql_in_snuba(subscription, snuba_query, snql_query, entity_subscrip
         )
 
     entity_key = get_entity_key_from_request(snql_query)
+
+    if in_random_rollout("snuba.snql.enable-orjson"):
+        post_body = orjson.dumps(body).decode("utf-8")
```
Comment (asottile-sentry) [inline src/sentry/snuba/tasks.py:257]: `.decode()` is slightly faster than `.decode('utf-8')` and the default is utf-8 in python 3

Comment (anonrig) [reply src/sentry/snuba/tasks.py]: Nice catch. You're absolutely right.

File: src/sentry/snuba/tasks.py [hunk 2]
```diff
@@ -250,14 +252,23 @@ def _create_snql_in_snuba(subscription, snuba_query, snql_query, entity_subscrip
...
+
+    if in_random_rollout("snuba.snql.enable-orjson"):
+        post_body = orjson.dumps(body).decode()
+    else:
+        post_body = json.dumps(body)
     response = _snuba_pool.urlopen(
         "POST",
         f"/{snuba_query.dataset}/{entity_key.value}/subscriptions",
-        body=json.dumps(body),
+        body=post_body,
```
Comment (asottile-sentry) [inline src/sentry/snuba/tasks.py:263]: and it looks like `body` can take bytes directly so the `.decode()` is just wasted cpu

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**